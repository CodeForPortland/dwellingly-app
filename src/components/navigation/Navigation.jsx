import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { injectIntl, intlShape } from 'react-intl';
import UserControls from '../user-controls/UserControls';
import NavigationContent from './NavigationContent';
import Icon from '../icon/Icon';

import './Navigation.scss';
import { ROLES } from '../../constants/constants';

import { dummyUser } from '../../data';

class Navigation extends Component {
  constructor(props) {
    super(props);

    this.handleToggleMenu = this.handleToggleMenu.bind(this);

    this.state = {
      showMenu: false,
      user: dummyUser
    };
  }

  handleToggleMenu() {
    this.setState(({ showMenu }) => ({ showMenu: !showMenu }));
  }

  render() {
    const { desktopOnly, intl } = this.props;
    const { showMenu, user } = this.state;
    if (user.role !== ROLES.PROPERTY_MANAGER) {
      return (
        <div className="navigation">
          {desktopOnly && (
            <div className="frontmatter">
              <NavigationContent intl={intl} type="desktop" user={user} />
              <UserControls type="desktop" />
            </div>
          )}
          {!showMenu &&
            !desktopOnly && (
              <div className="actions">
                <button
                  type="button"
                  aria-label="Show Menu"
                  className="menu__btn action action--right"
                  onClick={this.handleToggleMenu}>
                  <Icon icon="menu" />
                </button>
              </div>
            )}

          {showMenu &&
            !desktopOnly && (
              <NavigationContent
                handleToggleMenu={this.handleToggleMenu}
                intl={intl}
                user={user}
              />
            )}
        </div>
      );
    }

    return null;
  }
}

Navigation.propTypes = {
  desktopOnly: PropTypes.bool,
  intl: intlShape.isRequired
};

Navigation.defaultProps = {
  desktopOnly: false
};

export default injectIntl(Navigation);
