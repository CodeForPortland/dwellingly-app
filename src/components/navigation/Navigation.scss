@import "../../constants/constants.scss";

/* MENU */
.menu {
  height: 100vh;
  position: fixed;
  text-align: left;
  width: 100%;
  z-index: 100;

  &--desktop {
    display: none;
    left: 0;
    width: $width__nav;
  }

  a {
    @include flex_align(space-between);
    color: $color__white;
    // font-weight: 500;
    padding: 0.25em 1.5em;
  }

  &__bg {
    background-color: rgba(37, 37, 37, 0.7);
    height: 100vh;
    position: absolute;
    width: 100%;
  }

  &__groups {
    @include hidden_scrolling;
    background-image: $color__gradient--nav;
    color: $color__white;
    height: 100vh;
    margin: 0;
    position: absolute;
    right: 0;
    width: $width__nav;

    dl {
      margin: 0;
      padding: 3em 0;
    }

    dd {
      margin: 0 0 1em 0;
    }
  }

  &__close-btn {
    color: $color__white;
    font-size: 1.3125em;
    position: absolute;
    right: 1rem;
    top: 1rem;
  }

  &__logged-in-user-avatar {
    border: 1px solid $color__white;
    border-radius: 50%;
    height: 2.5rem;
    left: 1rem;
    overflow: hidden;
    position: absolute;
    top: 1rem;
    width: 2.5rem;

    img {
      width: 100%;
    }
  }

  &__label {
    padding: 1em 1.5em 0.5em 1.5em;

    .icon {
      font-size: 1.25em;
      margin-right: 0.5em;
    }
  }
}

@media only screen and (min-width: $mq--desktop) {
  .menu--desktop {
    display: block;
  }

  .menu--mobile,
  button.menu__btn {
    display: none;
  }
}
/* end menu */
