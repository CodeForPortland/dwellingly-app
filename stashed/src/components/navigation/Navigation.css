/* V A R I A B L E S */
/* end variables */
/* F U N C T I O N S */
/* end functions */
/* M I X I N S */
/* end mixins */
.navigation {
  background-color: #4ca1ff;
}

/* MENU */
.menu {
  height: 100vh;
  position: fixed;
  text-align: left;
  width: 100%;
  z-index: 100;
}

.menu--desktop {
  display: none;
  height: auto;
  position: relative;
  width: 15em;
}

.menu a {
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
      -ms-flex-pack: justify;
          justify-content: space-between;
  color: #ffffff;
  padding: 0.25em 1.5em;
}

.menu__bg {
  background-color: rgba(37, 37, 37, 0.7);
  height: 100vh;
  position: absolute;
  width: 100%;
}

.menu__groups {
  background-image: -webkit-gradient(linear, left top, left bottom, from(#4cb9ef), to(#6d8dff));
  background-image: linear-gradient(180deg, #4cb9ef 0%, #6d8dff 100%);
  -webkit-box-sizing: content-box;
          box-sizing: content-box;
  overflow-y: auto;
  -ms-overflow-style: none;
  color: #ffffff;
  height: 100vh;
  margin: 0;
  position: absolute;
  right: 0;
  width: 15em;
}

.menu__groups::-webkit-scrollbar {
  display: none;
}

.menu__groups dl {
  margin: 0;
  padding: 3em 0;
}

.menu__groups dd {
  margin: 0 0 1em 0;
}

.menu__close-btn {
  color: #ffffff;
  font-size: 1.3125em;
  position: absolute;
  right: 1rem;
  top: 1rem;
}

.menu__logged-in-user-avatar {
  border: 1px solid #ffffff;
  border-radius: 50%;
  height: 2.5rem;
  left: 1rem;
  overflow: hidden;
  position: absolute;
  top: 1rem;
  width: 2.5rem;
}

.menu__logged-in-user-avatar img {
  width: 100%;
}

.menu__label {
  padding: 1em 1.5em 0.5em 1.5em;
}

.menu__label .icon {
  font-size: 1.25em;
  margin-right: 0.5em;
}

@media only screen and (min-width: 1200px) {
  .menu--desktop {
    display: block;
  }
  .menu--desktop .menu__groups {
    background-image: none;
    height: auto;
    overflow-y: auto;
    position: relative;
  }
  .menu--mobile,
  button.menu__btn {
    display: none;
  }
}

/* end menu */
/*# sourceMappingURL=Navigation.css.map */