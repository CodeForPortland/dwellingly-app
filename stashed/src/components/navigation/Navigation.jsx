import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Cookies from 'universal-cookie';
import { connect } from 'react-redux';
import { injectIntl, intlShape } from 'react-intl';
import { ROLES } from '../../constants/constants';
import { getUserRoleString } from '../../utils';
// import UserControls from '../user-controls/UserControls';
import LogoutButton from '../login/LogoutButton';
import NavigationContent from './NavigationContent';
import Icon from '../icon/Icon';
import { clearUser } from '../../dux/user';

import './Navigation.scss';

class Navigation extends Component {
  constructor(props) {
    super(props);

    this.handleToggleMenu = this.handleToggleMenu.bind(this);
    this.clearUserCookie = this.clearUserCookie.bind(this);

    this.state = {
      showMenu: false
    };
  }

  handleToggleMenu() {
    this.setState(({ showMenu }) => ({ showMenu: !showMenu }));
  }

  clearUserCookie() {
    const cookies = new Cookies();
    this.props.clearUser();
    cookies.remove('messengerUser');
    cookies.remove('messengerUserId');
    cookies.remove('messengerUserRole');
  }

  render() {
    const { desktopOnly, intl, user } = this.props;
    let userRole;
    if (user) {
      userRole = getUserRoleString(user.role, ROLES);
    }
    const { showMenu } = this.state;
    if (userRole && userRole.isPropertyManager !== true) {
      return (
        <div className="navigation">
          {desktopOnly && (
            <div className="frontmatter">
              <NavigationContent
                intl={intl}
                type="desktop"
                userRole={userRole}
                user={user}
              />
              {/* <UserControls type="desktop" /> */}
            </div>
          )}
          {!showMenu &&
            !desktopOnly && (
              <div className="actions">
                <button
                  type="button"
                  aria-label="Show Menu"
                  className="menu__btn action action--right"
                  onClick={this.handleToggleMenu}>
                  <Icon icon="menu" />
                </button>
              </div>
            )}

          {showMenu &&
            !desktopOnly && (
              <NavigationContent
                handleToggleMenu={this.handleToggleMenu}
                intl={intl}
                user={user}
              />
            )}
          <LogoutButton clearCookie={this.clearUserCookie} />
        </div>
      );
    }

    return null;
  }
}

// const mapStateToProps = state => ({
//   user: state.user
// });

const mapDispatchToProps = dispatch => ({
  clearUser: () => dispatch(clearUser())
});

Navigation.propTypes = {
  user: PropTypes.shape({}),
  desktopOnly: PropTypes.bool,
  intl: intlShape.isRequired,
  clearUser: PropTypes.func.isRequired
};

Navigation.defaultProps = {
  user: null,
  desktopOnly: false
};

export default injectIntl(
  connect(
    null,
    mapDispatchToProps
  )(Navigation)
);
