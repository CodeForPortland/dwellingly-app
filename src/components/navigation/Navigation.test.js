import React from 'react';
import renderer from 'react-test-renderer';
import Navigation from './Navigation';

const NavigationComponentTest = (props) =>
<Navigation
{...defaultProps}
{...props}
/>;

describe('Navigation Component', () => {
  it('correctly renders Navigation component', () => {
    const NavigationComponent = renderer.create(NavigationComponentTest).toJSON();
    expect(NavigationComponent).toMatchSnapshot();
  });
});
