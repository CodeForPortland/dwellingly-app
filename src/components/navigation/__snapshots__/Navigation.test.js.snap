// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component correctly renders Navigation component 1`] = `null`;
